### PR TITLE
 Issue 846 - Add config file support to hzn

### PR DESCRIPTION
**Pull request by linggao**

### CODE REVIEW

File: cli/dev/hznenv.go [hunk 1]
```diff
@@ -1,33 +1,19 @@
...
 
 import (
-	"fmt"
+	"github.com/open-horizon/anax/cli/cliconfig"
 )
 
-const HZNENV_FILE = "hzn.cfg"
+const HZNENV_FILE = "hzn.json"
 
 // It creates a hzn.env file that contains the enviromental variables needed for this project.
```
Comment (bmpotter) [inline cli/dev/hznenv.go]: Comment is misleading because the code doesn't create hzn.env

File: cli/dev/image.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ const DEFAULT_BASE_OS_VALUE = "alpine:latest"
 const MAKE_FILE_CONTENT = `# Make targets for building the IBM example helloworld edge service
 
 # This imports the variables from horizon/hzn.cfg. You can ignore these lines, but do not remove them.
--include horizon/.hzn.cfg.tmp.mk
+-include PROJECT_DIR/.hzn.cfg.tmp.mk
```
Comment (bmpotter) [inline cli/dev/image.go]: Also a nit, but the file should now be called `.hzn.json.tmp.mk`

File: cli/hzn.go [hunk 3]
```diff
@@ -45,9 +46,13 @@ Environment Variables:
       to specify the organization ID'.
   HZN_EXCHANGE_USER_AUTH:  Default value for the 'hzn exchange -u' or 'hzn
       register -u' flag, in the form '[org/]user:pw'.
-  HZN_DONT_SUBST_ENV_VARS:  Set this to "1" to indicate that input json files
```
Comment (bmpotter) [inline cli/hzn.go]: Is the idea behind removing this from the help that we will still support it as a hidden option, but almost no one will need to know about it, so don't bother confusing them with it?

Comment (linggao) [reply cli/hzn.go]: You are absolutely right.

File: cli/dev/image.go (hunk 2 above)
Comment (linggao) [reply cli/dev/image.go]: done

File: cli/dev/hznenv.go (hunk 1 above)
Comment (linggao) [reply cli/dev/hznenv.go]: done

**Pull request merged**